server/src/test/java/org/opensearch/action/admin/cluster/snapshots/get/GetSnapshotsActionTests.java

```diff
@@ -0,0 +1,245 @@
+        when(clusterServiceMock.getClusterApplierService()).thenReturn(clusterApplierService);
+        Map<String, Repository.Factory> typesRegistry = Map.of(
+            RepositoriesServiceTests.TestRepository.TYPE,
+            RepositoriesServiceTests.TestRepository::new,
+            RepositoriesServiceTests.MeteredRepositoryTypeA.TYPE,
+            metadata -> new RepositoriesServiceTests.MeteredRepositoryTypeA(metadata, clusterServiceMock),
+            RepositoriesServiceTests.MeteredRepositoryTypeB.TYPE,
+            metadata -> new RepositoriesServiceTests.MeteredRepositoryTypeB(metadata, clusterServiceMock)
+        );
+
+        return typesRegistry;
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        ThreadPool.terminate(threadPool, 30, TimeUnit.SECONDS);
+        threadPool = null;
+        clusterService.close();
+        super.tearDown();
+    }
+
+    public void testRepoMissing() {
+        GetSnapshotsRequest repoMissingRequest = new GetSnapshotsRequest();
+        getSnapshotsAction.execute(null, repoMissingRequest, ActionListener.wrap(repoMissingResponse -> {
+            assertThrows(ActionRequestValidationException.class, () -> { repoMissingResponse.getSnapshots(); });
+        }, exception -> { assertTrue(exception.getMessage().contains("repository is missing")); }));
+
+        repoMissingRequest = new GetSnapshotsRequest().snapshots(snapshotsName1);
+
+        getSnapshotsAction.execute(null, repoMissingRequest, ActionListener.wrap(repoMissingResponse -> {
+            assertThrows(ActionRequestValidationException.class, () -> { repoMissingResponse.getSnapshots(); });
+        }, exception -> { assertTrue(exception.getMessage().contains("repository is missing")); }));
+
+        repoMissingRequest = new GetSnapshotsRequest().ignoreUnavailable(true);
+
+        getSnapshotsAction.execute(null, repoMissingRequest, ActionListener.wrap(repoMissingResponse -> {
+            assertThrows(ActionRequestValidationException.class, () -> { repoMissingResponse.getSnapshots(); });
+        }, exception -> { assertTrue(exception.getMessage().contains("repository is missing")); }));
+
+        repoMissingRequest = new GetSnapshotsRequest().verbose(true);
+
+        getSnapshotsAction.execute(null, repoMissingRequest, ActionListener.wrap(repoMissingResponse -> {
+            assertThrows(ActionRequestValidationException.class, () -> { repoMissingResponse.getSnapshots(); });
+        }, exception -> { assertTrue(exception.getMessage().contains("repository is missing")); }));
+
+        repoMissingRequest = new GetSnapshotsRequest().verbose(false);
+
+        getSnapshotsAction.execute(null, repoMissingRequest, ActionListener.wrap(repoMissingResponse -> {
+            assertThrows(ActionRequestValidationException.class, () -> { repoMissingResponse.getSnapshots(); });
+        }, exception -> { assertTrue(exception.getMessage().contains("repository is missing")); }));
```

Comment:
i dont think we need each request param combination here as you are testing for repo missing.